core/dbt/context/context_config.py

```diff
@@ -82,8 +82,9 @@ def get_config_dict(self, resource_type: NodeType) -> Dict[str, Any]:
 
 
 class BaseContextConfigGenerator(Generic[T]):
-    def __init__(self, active_project: RuntimeConfig):
+    def __init__(self, active_project: RuntimeConfig, language: Optional[ModelLanguage] = None):
```

Comment:
@gshank @ChenyuLInx Does this seem like the best way to do this? It seems odd to me that language would be the one element that needs to be passed separately.

Reply:
@devmessias Thanks for the contribution and engaging discussion, sorry we have been a behind on this patch.

I have the same feeling when reading this code.
I can see the appeal of having a proper PythonNode with default materialization being `table`.
But because this node is buried deep under layers functions, I don't think having this language passed through is ideal.
I would recommend going back to what Gerda initially proposed ([link](https://github.com/dbt-labs/dbt-core/commit/8f95385262d24ca68e3328e5dcdd7daf7061bc0a))

> In update_parsed_node_config check that the language is python and that materialization is not set in the config dict, and if both of those are true, set it in the config dict.

That would make this PR easier to follow. What do you think?

Reply:
Hi, thank you for your feedback

It’s been quite some time since I worked on this, but if memory serves me right, the approach you’re referring to is indeed what I implemented in this commit: https://github.com/dbt-labs/dbt-core/pull/8538/commits/8f95385262d24ca68e3328e5dcdd7daf7061bc0a. However, this change introduced another bug which was caught by a unit test, as documented here: [unit test issue comment](https://github.com/dbt-labs/dbt-core/pull/8538#issuecomment-1744871154)

Reply:
I believe the need to have a Python node is because the default materialization type in dbt is not supported for Python models (at least when I made this PR). This issue was not apparent before because it was being obscured by a materialization bug.

Reply:
I think this the unit test is failing because the config_dict generated by `config_dict = config.build_config_dict(patch_config_dict=patch_config_dict)` is what's being added to the parsed node in the end.
And I see why you have the current solution which is creating a place to define a default node.

I think passing the language through multiple layers of calls is not ideal, but if we want to do something better, we have two ways:
- do some large refactors to have this config consolidation function moved to a node and that would give us an easier way to define a default materialization on a python model node
- slightly twist the patch_config_dict(or create a new version of it) with any config that is different from the default default, and apply it. We should add some comments to it also. (I personally prefer this for the current case, and we can refactor later on if we see a pattern developing here).


So this would look like adding the following logic [here](https://github.com/dbt-labs/dbt-core/commit/8f95385262d24ca68e3328e5dcdd7daf7061bc0a#diff-0afffd02471ddca604b25dd73c0b0c411bafab7cb5a48f3d640a3cc9e613c7ceR327), and update the comment a bit.
```
if not patch_config_dict:
    patch_config_dict = {}
if parsed_node.resource_type == NodeType.Model and parsed_node.language == ModelLanguage.python:
    if "materialized" not in patch_config_dict:
        patch_config_dict["materialized"] = "table"
```

@devmessias @gshank what do you think?

Reply:
@devmessias are you still interested in making an update to this? No worries if not, we can take it across the finish line. Thanks either way for putting so much effort into this!

Reply:
Yes, I’m still interested even though I’ve switched jobs. Now it’s a matter of honor :laughing: . I will make the modifications this week and do a rebase.

Reply:
Hi @devmessias just wanted to check if you're still interested in completing this PR!